### DISCUSSION TITLE
C++

### DISCUSSION THREAD

**Discussion by ezamosch** [RussianInvestments/investAPI#97] [Q&A] [answered]
Добрый день.

С++ не поддерживается для gRPC протокола?
Скомпилировал protobuf файл, но доступных методов в CPP файлах не увидел.
![image](https://github.com/RussianInvestments/investAPI/assets/48468561/0ee26631-263c-4134-9ef2-024115ca062a)

Я так понимаю это список доступных языков?

**Comment by ezamosch** [comment]
![image](https://github.com/RussianInvestments/investAPI/assets/48468561/658e98dd-de8a-4174-9fa5-04d95214beaf)

**Comment by volkov-invest** [comment, marked as answer]
Добрый день!

Нет, это список опций, для c++ их нет: https://protobuf.dev/programming-guides/proto3/#packages

Сгенерировать код-обертку можно как [вручную ](https://grpc.io/docs/languages/cpp/quickstart/), так и воспользоваться [неофициальным ](https://github.com/samoilovv/TinkoffInvestSDK) sdk, официального пока нет.

Отмечу, что при ручной генерации может быть проблема с именованием namespace, вот решение - https://github.com/RussianInvestments/investAPI/issues/11
